app/src/main/report/components/Versions/VersionDetails.tsx

```diff
@@ -8,34 +8,41 @@ import {ParameterList} from "../Parameters/ParameterList";
 import {DataLinks} from "../Data/DataLinks";
 import {ResourceLinks} from "../Resources/ResourceLinks";
 import {ArtefactsList} from "../Artefacts/ArtefactsList";
-import { FileDownloadLink } from "../FileDownloadLink";
+import {FileDownloadLink} from "../FileDownloadLink";
+import {ReportVersionSwitcher} from "./ReportVersionSwitcher";
+import {IRouter} from "simple-react-router";
 
 const styles = require("../../styles/reports.css");
 
-export interface VersionProps extends RemoteContent {
-    versionDetails: Version,
-    report: string
+interface PublicProps {
+    onChangeVersion: (version: string) => void;
+}
+
+export interface VersionProps extends RemoteContent, PublicProps {
+    versionDetails: Version;
+    report: string;
+    otherVersions: string[];
 }
 
 export class VersionDetailsComponent extends RemoteContentComponent<VersionProps> {
     static getStores() {
         return [reportStore];
     }
 
-    static getPropsFromStores(): VersionProps {
-
+    static getPropsFromStores(props: Partial<VersionProps>): VersionProps {
         const s = reportStore.getState();
-
         return {
             versionDetails: s.versionDetails[s.currentVersion],
             report: s.currentReport,
-            ready: s.ready && s.versionDetails[s.currentVersion] != null
+            otherVersions: s.versions[s.currentReport],
+            ready: s.ready && s.versionDetails[s.currentVersion] != null,
+
+            onChangeVersion: props.onChangeVersion
         };
     }
 
-    renderContent(props: VersionProps) {
-        const p = props;
-        const url = `/reports/${p.report}/versions/${p.versionDetails.id}/all/`;
+    private renderTable(props: VersionProps) {
+        const url = `/reports/${props.report}/${props.versionDetails.id}/all/`;
```

Comment:
one failing integration test - this url has changed!

Reply:
Fixed